screenpipe-server/src/server.rs

```diff
@@ -863,6 +866,213 @@ async fn execute_raw_sql(
+#[derive(Serialize)]
+pub struct AddContentResponse {
+    pub success: bool,
+    pub message: Option<String>,
+}
+
+async fn add_frame_to_db(
+    state: &AppState,
+    frame: &FrameContent,
+    device_name: &str,
+) -> Result<(), anyhow::Error> {
+    let db = &state.db;
+
+    let frame_id = db.insert_frame(
+        &device_name,
+        Some(frame.timestamp.unwrap_or_else(Utc::now)),
+    ).await?;
+
+    if let Some(ocr_results) = &frame.ocr_results {
+        for ocr in ocr_results {
+            db.insert_ocr_text(
+                frame_id,
+                &ocr.text,
+                &ocr.text_json.as_deref().unwrap_or(""),
+                &frame.app_name.as_deref().unwrap_or(""),
+                &frame.window_name.as_deref().unwrap_or(""),
+                Arc::new(OcrEngine::default()), // Ideally could pass any str as ocr_engine since can be run outside of screenpipe
+                false
+            ).await?;
+        }
+    }
+
+    if let Some(tags) = &frame.tags {
+        db.add_tags(frame_id, TagContentType::Vision, tags.clone()).await?;
+    }
+
+    Ok(())
+}
+
+fn encode_frame_from_file_path(file_path: &str) -> Result<Vec<u8>, anyhow::Error> {
+    let image = image::open(file_path)?;
+    let mut buffer = Vec::new();
+    image.write_to(&mut std::io::Cursor::new(&mut buffer), ImageFormat::Png)?;
+    Ok(buffer)
+}
+
+async fn write_frames_to_video(
+    frames: &Vec<FrameContent>,
+    video_file_path: &str,
+    fps: f64,
+) -> Result<(), anyhow::Error> {
+    let mut ffmpeg_child = start_ffmpeg_process(video_file_path, fps).await?;
+    let mut ffmpeg_stdin = ffmpeg_child.stdin.take().expect("Failed to open stdin for FFmpeg");
+
+    for frame in frames {
+        let encoded_frame = encode_frame_from_file_path(&frame.file_path)?;
+        if let Err(e) = write_frame_to_ffmpeg(&mut ffmpeg_stdin, &encoded_frame).await {
+            error!("Failed to write frame to FFmpeg: {}", e);
+            return Err(e);
+        }
+    }
+
+    finish_ffmpeg_process(ffmpeg_child, Some(ffmpeg_stdin)).await;
+    Ok(())
+}
+
+async fn add_transcription_to_db(
+    state: &AppState,
+    transcription: &AudioTranscription,
+    device_name: &str,
+) -> Result<(), anyhow::Error> {
+    let db = &state.db;
+
+    let device = AudioDevice {
+        name: device_name.to_string(),
+        device_type: DeviceType::Input,
+    };
+
+    db.insert_audio_transcription(
+        None, // No associated audio chunk
+        &transcription.transcription,
+        -1,
+        &transcription.transcription_engine,
+        &device,
+    ).await?;
+
+    Ok(())
+}
+
+pub(crate) async fn add_to_database(
+    State(state): State<Arc<AppState>>,
+    Json(payload): Json<AddContentRequest>,
+) -> Result<JsonResponse<AddContentResponse>, (StatusCode, JsonResponse<Value>)> {
+
+    let device_name = payload.device_name.clone();
+    let mut success_messages = Vec::new();
+
+    match payload.content.content_type.as_str() {
+        "Frames" => {
```

Comment:
why is that uppercase?

seems like it should be snake case

Reply:
Yup sorry this was left over from a different way of doing this. Will fix!